dds/DCPS/MultiTopicDataReader_T.cpp

```diff
@@ -104,33 +102,33 @@ MultiTopicDataReader_T<Sample, TypedDataReader>::join(
       SampleInfo info;
       const ReturnCode_t ret = other_dri->read_instance_generic(other_data.ptr_,
         info, ih, READ_SAMPLE_STATE, ANY_VIEW_STATE, ALIVE_INSTANCE_STATE);
-      if (ret != RETCODE_OK && ret != RETCODE_NO_DATA) {
+      if (ret == DDS::RETCODE_NO_DATA || !info.valid_data) {
```

Comment:
This looks at the state of `info` even in the case of other (not "OK") return values.

Reply:
The `ret != RETCODE_OK` should be the first check.